packages/portal/backend/src/Factory.ts

```diff
@@ -76,24 +82,46 @@ export class Factory {
         //     return Factory.controller;
         // }
 
-        if (typeof ghController === 'undefined') {
+        if (typeof ghController === "undefined") {
             ghController = new GitHubController(GitHubActions.getInstance());
         } else {
             // really only for testing
             Log.trace("Factory::getCourseController() - using provided controller");
         }
         try {
+            const pluginPath = Config.getInstance().getProp(ConfigKey.plugin_fullpath);
+
+            Log.trace("Factory::getCourseController() - name: " + name + "; path: " + pluginPath);
+
             // NOTE: using require instead of import because file might not be present in forks
             // import complains about this, but require does not.
-            let plug: any;
-            if (name === 'classytest') {
-                Log.trace("Factory::getCourseController() - name: " + name + " - plug: DefaultCourseController");
-                plug = await require('./custom/DefaultCourseController'); // default for testing
-            } else {
-                // If a course wants to specialize the AdminView it should be in the file below.
-                // This is not required. But if it is added, it should never be pushed back to 'classy/master'
-                Log.trace("Factory::getCourseController() - name: " + name + " - plug: CustomCourseController");
-                plug = await require('./custom/CustomCourseController');
+            let plug: any = null;
+
+            try {
+                if (name !== Factory.TESTNAME && typeof pluginPath !== "undefined") {
+                    // If a course wants to specialize the AdminView it should be in the file below.
+                    // This is not required. But if it is added, it should never be pushed back to 'classy/master'
+                    Log.trace("Factory::getCourseController() - name: " + name + " - plug: CustomCourseController");
+                    plug = await require(pluginPath + "/plugin/src/backend/CustomCourseController");
```

Comment:
We have an extra `/plugin` folder here, which cannot be accounted for. 

![image](https://user-images.githubusercontent.com/7872295/116117608-47242b00-a671-11eb-97cb-2b5774e3fc50.png)

The `/src/` folder should be the root filepath after the pluginPath, should it not? 

ie. pluginPath = '/opt/classy-plugin/` + '/src/backend/CustomCourseController'.

Reply:
We would have to remove the `/plugin` path as part of the other files as well. 

Otherwise we end up with a duplicate plugin name. 

ie. /lead/up/to/classy-plugin/plugin/src/...

Let's just call it /lead/up/to/classy-plugin/src where `classy-plugin` is based on the repository name of the plugin that one clones.